xesmf/util.py

```diff
@@ -96,3 +150,197 @@ def grid_global(d_lon, d_lat):
+    """
+    return has_units_of_longitude(var)
+
+
+def is_latitude(var):
+    """Return True if variable is a latitude.
+
+    A variable is considered a latitude if its `units` are 'degrees_north' or a synonym.
+
+    Parameters
+    ----------
+    var: xr.Variable
+      Coordinate variable.
+
+    Return
+    ------
+    bool
+      True if variable is a latitude.
+    """
+    return has_units_of_latitude(var)
+
+
+def has_name_of_longitude(var):
+    key = "standard_name"
+    return var.attrs.get(key) == "longitude"
+
+
+def has_name_of_latitude(var):
+    key = "standard_name"
+    return var.attrs.get(key) == "latitude"
+
+
+def has_units_of_longitude(var):
+    return var.attrs.get("units") in ("degrees_east", "degree_east", "degree_E", "degrees_E", "degreeE", "degreesE")
+
+
+def has_units_of_latitude(var):
+    return var.attrs.get("units") in ("degrees_north", "degree_north", "degree_N", "degrees_N", "degreeN", "degreesN")
+
+
+def cf_lon_lat(ds):
+    """Return longitude and latitude.
+
+    Identify the longitude and latitude coordinates based on the value of the `long_name` attribute.
+
+    Parameters
+    ----------
+    ds: xr.Dataset
+      Dataset storing coordinate information.
+
+    Return
+    ------
+    xr.DataArray, xr.DataArray
+      Longitude and latitude coordinates.
+    """
+    if isinstance(ds, xr.Dataset):
+        x = list(ds.coords.values()) + list(ds.data_vars.values())
+    elif isinstance(ds, xr.DataArray):
+        x = ds.coords.values()
+    else:  # dict
+        x = ds.values()
+
+    lon = list(filter(is_longitude, x))
+    lat = list(filter(is_latitude, x))
+
+    msg = "No {0} coordinate found." \
+          "Identify the longitude variable by setting its `long_name` attribute to `{0}`."
+
+    if not lon:
+        raise ValueError(msg.format("longitude"))
+
+    if not lat:
+        raise ValueError(msg.format("latitude"))
+
+    return lon[0], lat[0]
+
+
+def cf_bnds(ds, coord):
+    """Return the coordinate boundaries for a given coordinate.
+
+    Identify the boundaries based on the value of the `bounds` attribute of the coordinate.
+    If coordinate has no `bounds` attribute, raise an error.
+
+    Parameters
+    ----------
+    ds: xr.Dataset
+      Dataset storing coordinate information.
+    coord: xr.DataArray
+      Coordinate whose boundaries should be returned.
+
+    Return
+    ------
+    xr.DataArray
+      The coordinate boundaries.
+    """
+    key = coord.attrs.get("bounds")
+    if key is None:
+        raise ValueError(f"No bounds found for {coord.name}."
+                         "Identify the longitude bounds by setting the `bounds` attribute of the longitude variable "
+                         "to an existing variable name.")
+    if key not in ds:
+        raise ValueError(f"The variable {key} identified as the bounds for {coord.name} is not in this dataset.")
+
+    return ds[key]
+
+
+def cf_lon_lat_bnds(ds):
+    """Return longitude and latitude boundaries.
```

Comment:
Maybe add a reminder about the array shapes.

Reply:
Done